skimage/_shared/fast_exp.pxd

```diff
@@ -0,0 +1,17 @@
+#cython: initializedcheck=False
+#cython: wraparound=False
+#cython: boundscheck=False
+
+cimport numpy as cnp
+from .fused_numerics cimport np_floats
+
+cdef extern from "fast_exp.h":
+    double fast_exp(double y) nogil
+    float fast_expf(float y) nogil
+
+
+cdef inline np_floats _fast_exp(np_floats x) nogil:
+    if np_floats is cnp.float32_t:
+        return fast_expf(x)
+    else:
+        return fast_exp(x)
```

Comment:
@rfezzani could you add some benchmarks on float32 and float64 volumes and check before and after this PR, and report back? I would like to be reassured that this extra branch (?) isn't actually resulting in greater runtime for float64...